### ISSUE TITLE
java.lang.RuntimeException: Couldn't greate generated-conf dir

### ISSUE THREAD

**Issue by salimane** [mesos/storm#70]
I'm getting the following on startup of the nimbus. anything I should have set ?
```shell
2015-11-29 18:00:07 o.a.s.z.s.ZooKeeperServer [INFO] Server environment:java.library.path=/usr/local/lib:/opt/local/lib:/usr/lib
2015-11-29 18:00:07 o.a.s.z.s.ZooKeeperServer [INFO] Server environment:java.io.tmpdir=/tmp
2015-11-29 18:00:07 o.a.s.z.s.ZooKeeperServer [INFO] Server environment:java.compiler=<NA>
2015-11-29 18:00:07 o.a.s.z.s.ZooKeeperServer [INFO] Server environment:os.name=Linux
2015-11-29 18:00:07 o.a.s.z.s.ZooKeeperServer [INFO] Server environment:os.arch=amd64
2015-11-29 18:00:07 o.a.s.z.s.ZooKeeperServer [INFO] Server environment:os.version=3.10.0-123.el7.x86_64
2015-11-29 18:00:07 o.a.s.z.s.ZooKeeperServer [INFO] Server environment:user.name=storm
2015-11-29 18:00:07 o.a.s.z.s.ZooKeeperServer [INFO] Server environment:user.home=/home/storm
2015-11-29 18:00:07 o.a.s.z.s.ZooKeeperServer [INFO] Server environment:user.dir=/opt/storm
2015-11-29 18:00:07 o.m.log [INFO] Logging to Logger[org.mortbay.log] via org.mortbay.log.Slf4jLog
2015-11-29 18:00:07 o.m.log [INFO] jetty-6.1.26
2015-11-29 18:00:07 o.m.log [INFO] Started SelectChannelConnector@0.0.0.0:46649
2015-11-29 18:00:07 s.m.MesosNimbus [INFO] Started HTTP server from which config for the MesosSupervisor's may be fetched. URL: http://idsp-mm-101.local:46649/generated-conf
2015-11-29 18:00:07 s.m.MesosNimbus [ERROR] Failed to prepare scheduler
java.lang.RuntimeException: Couldn't greate generated-conf dir
	at storm.mesos.MesosNimbus.initialize(MesosNimbus.java:162) [original-storm-0.9.5.jar:na]
	at storm.mesos.MesosNimbus.prepare(MesosNimbus.java:123) [original-storm-0.9.5.jar:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_91]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_91]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_91]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_91]
	at clojure.lang.Reflector.invokeMatchingMethod(Reflector.java:93) [clojure-1.5.1.jar:na]
	at clojure.lang.Reflector.invokeInstanceMethod(Reflector.java:28) [clojure-1.5.1.jar:na]
	at backtype.storm.daemon.nimbus$fn__3724$exec_fn__1103__auto____3725.invoke(nimbus.clj:896) [storm-core-0.9.5.jar:0.9.5]
	at clojure.lang.AFn.applyToHelper(AFn.java:163) [clojure-1.5.1.jar:na]
	at clojure.lang.AFn.applyTo(AFn.java:151) [clojure-1.5.1.jar:na]
	at clojure.core$apply.invoke(core.clj:617) [clojure-1.5.1.jar:na]
	at backtype.storm.daemon.nimbus$fn__3724$service_handler__3814.doInvoke(nimbus.clj:895) [storm-core-0.9.5.jar:0.9.5]
	at clojure.lang.RestFn.invoke(RestFn.java:421) [clojure-1.5.1.jar:na]
	at backtype.storm.daemon.nimbus$launch_server_BANG_.invoke(nimbus.clj:1152) [storm-core-0.9.5.jar:0.9.5]
	at backtype.storm.daemon.nimbus$_launch.invoke(nimbus.clj:1184) [storm-core-0.9.5.jar:0.9.5]
	at backtype.storm.daemon.nimbus.launch(Unknown Source) [storm-core-0.9.5.jar:0.9.5]
	at storm.mesos.MesosNimbus.main(MesosNimbus.java:98) [original-storm-0.9.5.jar:na]
2015-11-29 18:00:07 b.s.d.nimbus [ERROR] Error on initialization of server service-handler
java.lang.RuntimeException: java.lang.RuntimeException: Couldn't greate generated-conf dir
	at storm.mesos.MesosNimbus.prepare(MesosNimbus.java:137) [original-storm-0.9.5.jar:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_91]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_91]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_91]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_91]
	at clojure.lang.Reflector.invokeMatchingMethod(Reflector.java:93) ~[clojure-1.5.1.jar:na]
	at clojure.lang.Reflector.invokeInstanceMethod(Reflector.java:28) ~[clojure-1.5.1.jar:na]
	at backtype.storm.daemon.nimbus$fn__3724$exec_fn__1103__auto____3725.invoke(nimbus.clj:896) ~[storm-core-0.9.5.jar:0.9.5]
	at clojure.lang.AFn.applyToHelper(AFn.java:163) ~[clojure-1.5.1.jar:na]
	at clojure.lang.AFn.applyTo(AFn.java:151) ~[clojure-1.5.1.jar:na]
	at clojure.core$apply.invoke(core.clj:617) ~[clojure-1.5.1.jar:na]
	at backtype.storm.daemon.nimbus$fn__3724$service_handler__3814.doInvoke(nimbus.clj:895) [storm-core-0.9.5.jar:0.9.5]
	at clojure.lang.RestFn.invoke(RestFn.java:421) [clojure-1.5.1.jar:na]
	at backtype.storm.daemon.nimbus$launch_server_BANG_.invoke(nimbus.clj:1152) [storm-core-0.9.5.jar:0.9.5]
	at backtype.storm.daemon.nimbus$_launch.invoke(nimbus.clj:1184) [storm-core-0.9.5.jar:0.9.5]
	at backtype.storm.daemon.nimbus.launch(Unknown Source) [storm-core-0.9.5.jar:0.9.5]
	at storm.mesos.MesosNimbus.main(MesosNimbus.java:98) [original-storm-0.9.5.jar:na]
Caused by: java.lang.RuntimeException: Couldn't greate generated-conf dir
	at storm.mesos.MesosNimbus.initialize(MesosNimbus.java:162) [original-storm-0.9.5.jar:na]
	at storm.mesos.MesosNimbus.prepare(MesosNimbus.java:123) [original-storm-0.9.5.jar:na]
	... 16 common frames omitted
2015-11-29 18:00:07 b.s.util [ERROR] Halting process: ("Error on initialization")
java.lang.RuntimeException: ("Error on initialization")
	at backtype.storm.util$exit_process_BANG_.doInvoke(util.clj:325) [storm-core-0.9.5.jar:0.9.5]
	at clojure.lang.RestFn.invoke(RestFn.java:423) [clojure-1.5.1.jar:na]
	at backtype.storm.daemon.nimbus$fn__3724$service_handler__3814.doInvoke(nimbus.clj:895) [storm-core-0.9.5.jar:0.9.5]
	at clojure.lang.RestFn.invoke(RestFn.java:421) [clojure-1.5.1.jar:na]
	at backtype.storm.daemon.nimbus$launch_server_BANG_.invoke(nimbus.clj:1152) [storm-core-0.9.5.jar:0.9.5]
	at backtype.storm.daemon.nimbus$_launch.invoke(nimbus.clj:1184) [storm-core-0.9.5.jar:0.9.5]
	at backtype.storm.daemon.nimbus.launch(Unknown Source) [storm-core-0.9.5.jar:0.9.5]
	at storm.mesos.MesosNimbus.main(MesosNimbus.java:98) [original-storm-0.9.5.jar:na]
```

Thanks

**Comment by erikdw** [comment]
@salimane : the "generated-conf" dir is a new feature from #65's changes.  Probably a permission issue w.r.t. whatever the pwd/cwd of the MesosNimbus process is (which is specific to how you are starting it).  Which is pretty specific to how you're launching.  @brndnmtthws : what is the recommended way to run MesosNimbus w.r.t. location and permissions to overcome this issue?

**Comment by brndnmtthws** [comment]
Interesting. Nimbus currently tries to write to the CWD, but I see that's probably not the best way to do it. It might make more sense to use `storm.local.dir`. I'll follow-up with a PR.

**Comment by brndnmtthws** [comment]
PR sent. Give it a shot, and let me know if it works for you.

**Comment by erikdw** [comment]
@salimane : please try out @brndnmtthws 's change in #71 and let us know if it fixes [this issue](#70) for ya.

**Comment by erikdw** [comment]
#71's changes fixed this issue.
